cdk/cdk/cdk_stack.py

```diff
@@ -69,44 +88,75 @@ def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
             ),
         )
 
-        ## State Machine Steps
-        start_state = stepfunctions.Pass(self, "StartState")
-        discover_task = tasks.LambdaInvoke(
-            self, "Discover Granules Task", lambda_function=discover_lambda
+        ## CMR Workflow State Machine Steps
+        cmr_start_state = stepfunctions.Pass(self, "CMR Discovery StartState")
+        s3_start_state = stepfunctions.Pass(self, "S3 Discovery StartState")
+        cmr_discover_task = tasks.LambdaInvoke(
+            self, "CMR Discover Granules Task", lambda_function=cmr_discover_lambda
         )
+        s3_discover_task = tasks.LambdaInvoke(
+            self, "S3 Discover COGs Task", lambda_function=s3_discover_lambda
```

Comment:
I think this S3 task should be generic to file type - so this should just be "S3 Discover Task"